examples/nextjs/src/app/[[...slug]]/page.js

```diff
@@ -1,57 +1,62 @@
 import { MyPage } from "@/components/my-page";
-
-import { handleVanityUrlRedirect } from "../utils";
-import { dotcmsClient } from "@dotcms/client";
-
-const client = dotcmsClient.init({
-    dotcmsUrl: process.env.NEXT_PUBLIC_DOTCMS_HOST,
-    authToken: process.env.DOTCMS_AUTH_TOKEN,
-    siteId: "59bb8831-6706-4589-9ca0-ff74016e02b2",
-    requestOptions: {
-        // In production you might want to deal with this differently
-        cache: "no-cache",
-    },
-});
-
+import { ErrorPage } from "@/components/error";
+
+import { handleVanityUrlRedirect } from "@/utils/vanityUrlHandler";
+import { client, getRequestParams } from "@/utils/dotcmsClient";
+
+/**
+ * Generate metadata
+ *
+ * @export
+ * @param {*} { params, searchParams }
+ * @return {*}
+ */
 export async function generateMetadata({ params, searchParams }) {
-    const requestData = {
-        path: params?.slug ? params.slug.join("/") : "index",
-        language_id: searchParams.language_id,
-        "com.dotmarketing.persona.id":
-            searchParams["com.dotmarketing.persona.id"] || "",
-        mode: searchParams.mode,
-        variantName: searchParams["variantName"],
-    };
-
-    const data = await client.page.get(requestData);
-
-    return {
-        title: data.entity.page.friendlyName || data.entity.page.title,
-    };
+    const requestData = getRequestParams({
+        params,
+        searchParams,
+        defaultPath: "/",
+    });
+
+    try {
+        const data = await client.page.get(requestData);
+        const page = data.entity?.page;
+        const title = page?.friendlyName || page?.title;
+
+        return {
+            title,
+        };
+    } catch (e) {
+        return {
+            title: "not found",
+        };
+    }
 }
 
 export default async function Home({ searchParams, params }) {
-    const requestData = {
-        path: params?.slug ? params.slug.join("/") : "index",
-        language_id: searchParams.language_id,
-        "com.dotmarketing.persona.id":
-            searchParams["com.dotmarketing.persona.id"] || "",
-        mode: searchParams.mode,
-        variantName: searchParams["variantName"],
-    };
-
-    const data = await client.page.get(requestData);
-    const nav = await client.nav.get({
-        path: "/",
-        depth: 2,
-        languageId: searchParams.language_id,
+    const requestData = getRequestParams({
+        params,
+        searchParams,
+        defaultPath: "/",
     });
 
-    const { vanityUrl } = data.entity;
-
-    if (vanityUrl) {
-        handleVanityUrlRedirect(vanityUrl);
+    try {
+        const data = await client.page.get(requestData);
+        const nav = await client.nav.get({
+            path: "/",
+            depth: 2,
+            languageId: searchParams.language_id,
+        });
+        const { vanityUrl } = data?.entity;
```

Comment:
I think I pointed this out in a PR of @KevinDavilaDotCMS 

But we should be careful with this destructuring. This can lead to false positive errors in this case that we have a try/catch

But it's a matter of testing what happens

Reply:
That's why I use `try/catch`. However, if te entity is `null/undefine` we now throw an error in the `SDK` so it should not be a problem using the destructuring there.